internal/svcmanager/svcmanager.go

```diff
@@ -49,35 +49,46 @@ func (m *Manager) Start() error {
 }
 
 func (m *Manager) WaitWithContext(ctx context.Context) error {
+	defer profile.Measure("svcmanager:WaitWithContext", time.Now())
+
+	interrupt := false
 	waitDone := make(chan struct{})
 	var err error
+	
 	go func() {
-		err = m.Wait()
-		waitDone <- struct{}{}
+		defer func() { waitDone <- struct{}{} }()
+
+		logging.Debug("Waiting for state-svc")
+		for try := 1; try <= 10; try++ {
+			if interrupt {
+				return
+			}
+
+			logging.Debug("Attempt %d", try)
+			if m.Ready() {
```

Comment:
I think we can make `Ready` dependent on the global `ctx`. Then we could just interrupt it with a cancel or timeout.  That seems to be the correct solution here.  It might even make the go func obsolete.

Reply:
This works for me.  But what I had in mind was to replace lines 57 to 65 with this

```
		if m.Ready(ctx) {
			return nil
		}
		select {
			case <- time.After(250 * time.Millisecond):
			case <- ctx.Done():
					return nil
		}
```